rogue-thi-app/pages/timetable.jsx

```diff
@@ -175,6 +177,61 @@ export default function Timetable () {
     load(week[0])
   }, [router, timetable, focusedEntry, isDetailedData, week, fetchedWeek])
 
+  if (Object.keys(roomAvailabilityList).length === 0) {
+    loadRoomAvailability()
+  }
```

Comment:
This has to be wrapped inside useEffefct() because currently this functions initially gets called on the Server for SSR. Depending on how you configure the useEffect dependencies, you can get rid of the check if the `roomAvailabilityList` is empty. Because the current implementation only is intended for the current date (which makes sense), you can ignore the week dependencies, etc. so that the `loadRoomAvailability` only gets called once on load.